server/src/main/java/org/apache/druid/catalog/model/Columns.java

```diff
@@ -34,58 +35,72 @@ public class Columns
 {
   public static final String TIME_COLUMN = "__time";
 
-  public static final String VARCHAR = "VARCHAR";
-  public static final String BIGINT = "BIGINT";
-  public static final String FLOAT = "FLOAT";
-  public static final String DOUBLE = "DOUBLE";
-  public static final String TIMESTAMP = "TIMESTAMP";
+  public static final String STRING = ValueType.STRING.name();
+  public static final String LONG = ValueType.LONG.name();
+  public static final String FLOAT = ValueType.FLOAT.name();
+  public static final String DOUBLE = ValueType.DOUBLE.name();
+
+  public static final String SQL_VARCHAR = "VARCHAR";
+  public static final String SQL_BIGINT = "BIGINT";
+  public static final String SQL_FLOAT = "FLOAT";
+  public static final String SQL_DOUBLE = "DOUBLE";
+  public static final String SQL_TIMESTAMP = "TIMESTAMP";
 
   public static final Set<String> NUMERIC_TYPES =
-      ImmutableSet.of(BIGINT, FLOAT, DOUBLE);
+      ImmutableSet.of(LONG, FLOAT, DOUBLE);
   public static final Set<String> SCALAR_TYPES =
-      ImmutableSet.of(TIMESTAMP, VARCHAR, BIGINT, FLOAT, DOUBLE);
+      ImmutableSet.of(STRING, LONG, FLOAT, DOUBLE);
 
   public static final Map<String, ColumnType> SQL_TO_DRUID_TYPES =
       new ImmutableMap.Builder<String, ColumnType>()
-        .put(TIMESTAMP, ColumnType.LONG)
-        .put(BIGINT, ColumnType.LONG)
-        .put(FLOAT, ColumnType.FLOAT)
-        .put(DOUBLE, ColumnType.DOUBLE)
-        .put(VARCHAR, ColumnType.STRING)
-        .build();
+          .put(SQL_BIGINT, ColumnType.LONG)
+          .put(SQL_VARCHAR, ColumnType.STRING)
+          .build();
 
   public static final Map<ColumnType, String> DRUID_TO_SQL_TYPES =
       new ImmutableMap.Builder<ColumnType, String>()
-      .put(ColumnType.LONG, BIGINT)
-      .put(ColumnType.FLOAT, FLOAT)
-      .put(ColumnType.DOUBLE, DOUBLE)
-      .put(ColumnType.STRING, VARCHAR)
+      .put(ColumnType.LONG, SQL_BIGINT)
+      .put(ColumnType.FLOAT, SQL_FLOAT)
+      .put(ColumnType.DOUBLE, SQL_DOUBLE)
+      .put(ColumnType.STRING, SQL_VARCHAR)
       .build();
 
   private Columns()
   {
   }
 
-  public static boolean isTimestamp(String type)
-  {
-    return TIMESTAMP.equalsIgnoreCase(type.trim());
-  }
-
   public static boolean isScalar(String type)
   {
     return SCALAR_TYPES.contains(StringUtils.toUpperCase(type.trim()));
   }
 
-  public static ColumnType druidType(String sqlType)
+  public static ColumnType druidType(ColumnSpec spec)
   {
-    if (sqlType == null) {
+    if (isTimeColumn(spec.name())) {
+      return ColumnType.LONG;
+    }
+    String dataType = spec.dataType();
+    if (dataType == null) {
       return null;
     }
-    ColumnType druidType = SQL_TO_DRUID_TYPES.get(StringUtils.toUpperCase(sqlType));
+    ColumnType druidType = SQL_TO_DRUID_TYPES.get(StringUtils.toUpperCase(dataType));
```

Comment:
It looks like we changed the dataType of columnSpec to be the druid type instead of the sql typ now, so should we take the type as is now instead of looking up mapping here?

Reply:
It checks if it's a SQL type first and attempts to convert, and then does `ColumnType.fromString(dataType)` if not to treat it as a Druid native type